torch/distributed/_functional_collectives_impl.py

```diff
@@ -201,6 +201,17 @@ def _broadcast(self, src, tag, ranks, group_size):
     return inplace_tensor
 
 
+def _scatter(self, scatter_list, src, tag, ranks, group_size):
```

Comment:
I think we might want to implement this in the native approach now?

i.e. in https://github.com/pytorch/pytorch/blob/main/torch/csrc/distributed/c10d/Functional.cpp

cc @yifuwang could you also give some pointer about how to implement a inductor IR on a collective now?

Reply:
@wanchaol do we need both the native and python implementations?

Looking at https://github.com/pytorch/pytorch/blob/0cc60a05da2fed127b66f375cfc06252089b558a/torch/distributed/_functional_collectives.py#L143-L146

Reply:
We don't need to implement it in Python. We are removing the Python impls once we get everyone off of them.

Reply:
leaving it for now since it's useful to check correctness but can remove if you have a strong opinion otherwise